addons/sale_expense/tests/test_sale_expense.py

```diff
@@ -97,3 +98,52 @@ def test_sale_expense(self):
         # both expenses should be invoiced
         inv = so._create_invoices()
         self.assertEqual(inv.amount_untaxed, 621.5 + (prod_exp_2.list_price * 100.0), 'Sale Expense: invoicing of expense is wrong')
+
+    def test_expense_multi_id_analytic_distribution(self):
+        """
+        Test conversion of analytic_distribution dict into account numbers when a hr.expense.sheet containing
+        a hr.expense with an analytic_distribution having 2+ account ids
+        """
+        product = self.env['product.product'].create({
+            'name': 'test product',
+            'can_be_expensed': True,
+            'detailed_type': 'service',
+            'invoice_policy': 'order',
+            'list_price': 100,
+            'expense_policy': 'cost',
+        })
+
+        so = self.env['sale.order'].create({
+            'partner_id': self.partner_a.id,
+            'partner_invoice_id': self.partner_a.id,
+            'partner_shipping_id': self.partner_a.id,
+            'order_line': [Command.create({
+                'product_id': self.product_b.id,
+            })],
+        })
+        so.action_confirm()
+        so._create_invoices()
+
+        expense_sheet = self.env['hr.expense.sheet'].create({
+            'name': 'Test Expense Report',
+            'employee_id': self.expense_employee.id,
+            'company_id': self.company_data['company'].id,
+            'expense_line_ids': [Command.create({
+                'employee_id': self.expense_employee.id,
+                'can_be_reinvoiced': True,
+                'product_id': product.id,
+                'total_amount_currency': 1000.00,
+                'date': Date.today(),
+                'company_id': self.company_data['company'].id,
+                'currency_id': self.company_data['currency'].id,
+                'analytic_distribution': {f'{self.analytic_account_1.id}, {self.analytic_account_2.id}': 100},
```

Comment:
```suggestion
                'currency_id': self.company_data['currency'].id,
                'analytic_distribution': {f'{self.analytic_account_1.id}, {self.analytic_account_2.id}': 100, f'{self.analytic_account_3.id}': 40},
```
but prettier (and with a new analytic account)